lib/project-data.ts

```diff
@@ -30,7 +30,16 @@ export class ProjectData implements IProjectData {
 	public projectDir: string;
 	public platformsDir: string;
 	public projectFilePath: string;
-	public projectId: string;
+	public projectIdentifiers: IProjectIdentifier;
+	get projectId(): string {
+		this.warnProjectId();
+		return this.projectIdentifiers.ios;
+	}
+	set projectId(identifier: string) {
+		this.warnProjectId();
+		this.projectIdentifiers.ios = identifier;
```

Comment:
is there a validation for the identifier somewhere before this line? (check if valid ios identifier)

Reply:
No the validation happens at a later stage.

Reply:
I think validation should happen as early as possible, but it's up to you.